dfusion_rust_core/src/models/account_state.rs

```diff
@@ -73,9 +73,11 @@ impl AccountState {
         let buy_amounts = results.executed_buy_amounts;
         let sell_amounts = results.executed_sell_amounts;
 
-        for (i, order) in orders.iter().enumerate() {
-            self.increment_balance(order.buy_token, order.account_id, buy_amounts[i]);
-            self.decrement_balance(order.sell_token, order.account_id, sell_amounts[i]);
+        for (i, execution_amounts) in buy_amounts.iter().zip(sell_amounts.iter()).enumerate() {
```

Comment:
nit: maybe easier to just iterate by index i and not zip onto the other amounts?

Reply:
Ya, I could do that.

Reply:
The reason I chose to do this is because I find there is a lack of symmetry if I loop over one of `buy_amounts.len()` or `sell_amounts.len()`. I am not fond of this lopsidedness.